### PR TITLE
Add ability to search in current flow

### LINKED ISSUE #3419: feature: Allow search to be limited to the current flow
### Description

Restricting to the current flow is useful to keep the search scoped.

### PR DESCRIPTION
**Pull request by Steve-Mcl**
fixes #3419

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

Allow user to search current flow via the filter `is:thisflow`

## Checklist
- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by knolleary** [comment]
For the filter, I'm not sure it's an `is:XYZ` type filter. I suggest it should be `flow:current` - as that opens other possible options for filters related to the flow.

**Comment by Steve-Mcl** [comment]
> I suggest it should be flow:current - as that opens other possible options for filters related to the flow.

@knolleary 

Nick, What other features do you imagine? Like `flow:ID` or `"flow:"flow name"`? Knowing what you are thinking might influence the design. 


Ps, Not sure about "flow:current". Perhaps "flow:this" or "flow:active" would be a better term?

**Comment by Steve-Mcl** [comment]
Went with "flow:active" - hope thats ok (easy to change if not)

Also, the way it is implemented, you can search with in the active flow (`flow:active `) or another flow (`flow:1234abcd4567efgh `) or multiple flows (`flow:1234abcd4567efgh flow:9876abcd9876efgh `) 

Ready for review :)

File: packages/node_modules/@node-red/editor-client/locales/en-US/editor.json [hunk 1]
```diff
@@ -907,7 +907,8 @@
             "uknownNodes": "Unknown nodes",
             "unusedSubflows": "Unused subflows",
             "hiddenFlows": "Hidden flows",
-            "modifiedNodes": "Modified nodes and flows"
+            "modifiedNodes": "Modified nodes and flows",
+            "thisFlow": "This flow"
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/locales/en-US/editor.json:911]:
```suggestion
            "thisFlow": "Current flow"
```

File: packages/node_modules/@node-red/editor-client/src/js/ui/search.js [hunk 2]
```diff
@@ -122,19 +121,24 @@ RED.search = (function() {
...
-
-        var hasFlags = Object.keys(flags).length > 0;
-
+        val = extractValue(val,"flow",flags);// flow:active or flow:<flow-id>
+        val = extractValue(val,"uses",flags);// uses:<node-id>
         val = val.trim();
-
+        var hasFlags = Object.keys(flags).length > 0;
+        if (flags.flow && flags.flow.indexOf("active") >= 0) {
+            let idx = flags.flow.indexOf("active");
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/search.js:129]:
```suggestion
        if (flags.flow && flags.flow.indexOf("current") >= 0) {
            let idx = flags.flow.indexOf("current");
```

File: packages/node_modules/@node-red/editor-client/src/js/ui/search.js [hunk 3]
```diff
@@ -592,8 +601,8 @@ RED.search = (function() {
             {label:RED._("search.options.uknownNodes"), value: "type:unknown"},
             {label:RED._("search.options.unusedSubflows"), value:"is:subflow is:unused"},
             {label:RED._("search.options.hiddenFlows"), value:"is:hidden"},
+            {label:RED._("search.options.thisFlow"), value:"flow:active"},
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/search.js:604]:
```suggestion
            {label:RED._("search.options.thisFlow"), value:"flow:current"},
```

**Review by knolleary — COMMENTED** [review]
I would like to stick with `flow:current`.

We already use 'current flow' when talking about the Debug node filter - so this ought to be consistent with that.

**Comment by Steve-Mcl** [comment]
> I would like to stick with flow:current
> We already use 'current flow' 

Makes sense.

All done Nick. @knolleary

**Pull request merged**